staticcheck: too many warnings in travis-ci

via https://travis-ci.org/coreos/etcd/jobs/219525242
```
Checking staticcheck...
egrep: auth/store_test.go:640:4: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
auth/store_test.go:640:4: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
contrib/raftexample/raftexample_test.go:105:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
e2e/ctl_v3_watch_test.go:72:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
e2e/etcd_release_upgrade_test.go:156:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
etcdctl/ctlv3/command/elect_command.go:76:36: os.Kill cannot be trapped (did you mean syscall.SIGTERM?) (SA1016)
...
```

--- a/etcdctl/ctlv3/command/elect_command.go
+++ b/etcdctl/ctlv3/command/elect_command.go
@@ -18,6 +18,7 @@ import (
 	"errors"
 	"os"
 	"os/signal"
+	"syscall"
 
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/clientv3/concurrency"
@@ -73,7 +74,7 @@ func observe(c *clientv3.Client, election string) error {
 
 	donec := make(chan struct{})
 	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, os.Interrupt, os.Kill)
+	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
 	go func() {
 		<-sigc
 		cancel()
@@ -107,7 +108,7 @@ func campaign(c *clientv3.Client, election string, prop string) error {
 
 	donec := make(chan struct{})
 	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, os.Interrupt, os.Kill)
+	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
 	go func() {
 		<-sigc
 		cancel()

--- a/etcdctl/ctlv3/command/lock_command.go
+++ b/etcdctl/ctlv3/command/lock_command.go
@@ -18,6 +18,7 @@ import (
 	"errors"
 	"os"
 	"os/signal"
+	"syscall"
 
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/clientv3/concurrency"
@@ -57,7 +58,7 @@ func lockUntilSignal(c *clientv3.Client, lockname string) error {
 	// unlock in case of ordinary shutdown
 	donec := make(chan struct{})
 	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, os.Interrupt, os.Kill)
+	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
 	go func() {
 		<-sigc
 		cancel()

--- a/main_test.go
+++ b/main_test.go
@@ -29,7 +29,7 @@ func TestMain(t *testing.T) {
 	}
 
 	notifier := make(chan os.Signal, 1)
-	signal.Notify(notifier, syscall.SIGINT, syscall.SIGTERM, syscall.SIGKILL)
+	signal.Notify(notifier, syscall.SIGINT, syscall.SIGTERM)
 	go main()
 	<-notifier
 }

--- a/pkg/netutil/netutil_test.go
+++ b/pkg/netutil/netutil_test.go
@@ -150,7 +150,10 @@ func TestURLsEqual(t *testing.T) {
 		"second.com":  "10.0.11.2",
 	}
 	resolveTCPAddr = func(network, addr string) (*net.TCPAddr, error) {
-		host, port, err := net.SplitHostPort(addr)
+		host, port, herr := net.SplitHostPort(addr)
+		if herr != nil {
+			return nil, herr
+		}
 		if _, ok := hostm[host]; !ok {
 			return nil, errors.New("cannot resolve host.")
 		}

--- a/test
+++ b/test
@@ -281,7 +281,7 @@ function fmt_pass {
 		if [ -n "${staticcheckResult}" ]; then
 			# TODO: resolve these after go1.8 migration
 			# See https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck
-			STATIC_CHECK_MASK="SA(1016|1019|2002)"
+			STATIC_CHECK_MASK="SA(1019|2002)"
 			if echo "${staticcheckResult}" | egrep -v "$STATIC_CHECK_MASK"; then
 				echo -e "staticcheck checking ${path} failed:\n${staticcheckResult}"
 				exit 255

--- a/test
+++ b/test
@@ -250,16 +250,15 @@ function fmt_pass {
 	if which gosimple >/dev/null; then
 		echo "Checking gosimple..."
 		gosimpleResult=`gosimple ${STATIC_ANALYSIS_PATHS} 2>&1 || true`
-		if [ ! -n "${gosimpleResult}" ]; then
-			continue
-		fi
-		# TODO: resolve these after go1.8 migration
-		SIMPLE_CHECK_MASK="S(1024)"
-		if egrep -v "$SIMPLE_CHECK_MASK" "${gosimpleResult}"; then
-			echo -e "gosimple checking ${path} failed:\n${gosimpleResult}"
-			exit 255
-		else
-			echo -e "gosimple warning:\n${gosimpleResult}"
+		if [ -n "${gosimpleResult}" ]; then
+			# TODO: resolve these after go1.8 migration
+			SIMPLE_CHECK_MASK="S(1024)"
+			if echo "${gosimpleResult}" | egrep -v "$SIMPLE_CHECK_MASK"; then
+				echo -e "gosimple checking ${path} failed:\n${gosimpleResult}"
+				exit 255
+			else
+				echo -e "gosimple warning:\n${gosimpleResult}"
+			fi
 		fi
 	else
 		echo "Skipping gosimple..."
@@ -279,17 +278,17 @@ function fmt_pass {
 	if which staticcheck >/dev/null; then
 		echo "Checking staticcheck..."
 		staticcheckResult=`staticcheck ${STATIC_ANALYSIS_PATHS} 2>&1 || true`
-		if [ ! -n "${staticcheckResult}" ]; then
-			continue
-		fi
-		# TODO: resolve these after go1.8 migration
-		# See https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck
-		STATIC_CHECK_MASK="SA(1016|1019|2002)"
-		if egrep -v "$STATIC_CHECK_MASK" "${staticcheckResult}"; then
-			echo -e "staticcheck checking ${path} failed:\n${staticcheckResult}"
-			exit 255
-		else
-			echo -e "staticcheck warning:\n${staticcheckResult}"
+		if [ -n "${staticcheckResult}" ]; then
+			# TODO: resolve these after go1.8 migration
+			# See https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck
+			STATIC_CHECK_MASK="SA(1016|1019|2002)"
+			if echo "${staticcheckResult}" | egrep -v "$STATIC_CHECK_MASK"; then
+				echo -e "staticcheck checking ${path} failed:\n${staticcheckResult}"
+				exit 255
+			else
+				suppressed=`echo "${staticcheckResult}" | sed 's/ /\n/g' | grep "(SA" | sort | uniq -c`
+				echo -e "staticcheck suppressed warnings:\n${suppressed}"
+			fi
 		fi
 	else
 		echo "Skipping staticcheck..."

